frontend/pages/hosts/details/HostDetailsPage/modals/ScriptModalGroup/ScriptModalGroup.tsx

```diff
@@ -0,0 +1,200 @@
+import DeleteScriptModal from "pages/ManageControlsPage/Scripts/components/DeleteScriptModal";
+import RunScriptDetailsModal from "pages/DashboardPage/cards/ActivityFeed/components/RunScriptDetailsModal";
+import RunScriptModal from "../RunScriptModal";
+
+interface IScriptsProps {
+  currentUser: IUser | null;
+  host: IHost;
+  onCloseScriptModalGroup: () => void;
+  runScriptRequested: boolean;
+  setRunScriptRequested: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+type ScriptGroupModals =
+  | "run-script"
+  | "view-script"
+  | "run-script-details"
+  | "delete-script"
+  | null;
+
+const ScriptModalGroup = ({
+  currentUser,
+  host,
+  onCloseScriptModalGroup,
+  runScriptRequested,
+  setRunScriptRequested,
+}: IScriptsProps) => {
+  const [previousModal, setPreviousModal] = useState<ScriptGroupModals>(null);
+  const [currentModal, setCurrentModal] = useState<ScriptGroupModals>(
+    "run-script"
+  );
+  const [runScriptTablePage, setRunScriptTablePage] = useState(0);
+  const [selectedScriptId, setSelectedScriptId] = useState<number | undefined>(
+    undefined
+  );
+  const [selectedExecutionId, setSelectedExecutionId] = useState<
+    string | undefined
+  >(undefined);
+  const [selectedScriptDetails, setSelectedScriptDetails] = useState<
+    IHostScript | undefined
+  >(undefined);
+
+  // Almost everything from this is needed on RunScript.tsx modal
+  // except refetch is used multiple places
+  const {
+    data: runScriptTableResponse,
+    isError,
+    isLoading,
+    isFetching,
+    refetch: refetchHostScripts,
+  } = useQuery<
+    IHostScriptsResponse,
+    IApiError,
+    IHostScriptsResponse,
+    IHostScriptsQueryKey[]
+  >(
+    [
+      {
+        scope: "host_scripts",
+        host_id: host.id,
+        page: runScriptTablePage,
+        per_page: 10,
+      },
+    ],
+    ({ queryKey }) => scriptsAPI.getHostScripts(queryKey[0]),
+    {
+      refetchOnWindowFocus: false,
+      retry: false,
+      staleTime: 3000,
+      onSuccess: () => {
+        setRunScriptRequested(false);
+      },
+    }
+  );
+
+  // Note: Script metadata and script content require two separate API calls
+  // Source: https://fleetdm.com/docs/rest-api/rest-api#example-get-script
+  // So to get script name, we pass it into this modal instead of another API call
+  // If in future iterations we want more script metadata, call scriptAPI.getScript()
+  // and consider refactoring .getScript to return script content as well
+  const {
+    data: selectedScriptContent,
+    error: isSelectedScriptContentError,
+    isLoading: isLoadingSelectedScriptContent,
+  } = useQuery<any, Error>(
+    ["scriptContent", selectedScriptId],
+    () => scriptsAPI.downloadScript(selectedScriptId!),
+    {
+      refetchOnWindowFocus: false,
+      enabled: !!selectedScriptId,
+    }
+  );
+
+  useEffect(() => {
+    if (runScriptRequested) {
+      setCurrentModal("run-script");
+    }
+  }, [runScriptRequested]);
```

Comment:
I think we can remove this if we remove runScriptRequested prop and change some of the children modals.